mono/utils/lifo-semaphore.c

```diff
@@ -0,0 +1,91 @@
+#include <mono/utils/lifo-semaphore.h>
+
+LifoSemaphore *
+mono_lifo_semaphore_init (void)
+{
+	LifoSemaphore *semaphore;
+
+	semaphore = g_new0 (LifoSemaphore, 1);
+	if (semaphore == NULL)
+		return NULL;
+
+	mono_coop_mutex_init (&semaphore->mutex);
+
+	return semaphore;
+}
+
+void
+mono_lifo_semaphore_delete (LifoSemaphore *semaphore)
+{
+	g_assert (semaphore->head == NULL);
+	mono_coop_mutex_destroy (&semaphore->mutex);
+	g_free (semaphore);
+}
+
+int32_t
+mono_lifo_semaphore_timed_wait (LifoSemaphore *semaphore, int32_t timeout_ms)
+{
+	LifoSemaphoreWaitEntry wait_entry = { NULL, 0, NULL, NULL };
```

Comment:
You could shorten the initialize to merely `{ 0 }`, unless maybe there is a union in there, maybe.

Reply:
I thought this causes issues on MSVC, but maybe I'm misremembering. Will try.

Reply:
Guess not!